### ISSUE TITLE
Force on DLSS Frame Generation and Ray Reconstruction

### ISSUE THREAD

**Issue by avelev99** [emoose/DLSSTweaks#144]
Enhance DLSSTweaks by allowing users to force-enable DLSS Frame Generation™ and DLSS Ray Reconstruction™. Many games don't have FG and RR by default and while visual bugs are more than likely to occur having a force on function for the technology can help debug it, the game and in some cases improve the user experience by boosting the performance of the game.

Proposed Changes
1. Add option to force DLSS Frame Generation for smoother visuals.
2. Include functionality to activate DLSS Ray Reconstruction for improved lighting and reflections.

Benefits
- Provides greater control over DLSS settings.
- Boosts graphical fidelity and performance in games.

**Comment by d0x360** [comment]
I could be incorrect about this but I think frame gen need to be supported in the games rendering pipeline to work.  If it not you would have to grab the frame buffer and run a separate far less accurate interpolation method then use the same app to send that frame to the display. Aside from looking significantly worse it would also perform worse than FG while significantly increasing latency.

As for ray reconstruction... That is being run much like dlss for rays (but not).  In RT they are generally shooting the ray (probe) into the scene and having it bounce 1 to very few times. With path tracing it usually is more than 1 probe per pixel and will also randomly  bounce  pixel to pixel until it hits a light source but they can't set a bounce limit it is always significantly higher than regular RT. 

Knowing the difference in methods briefly explains why RT & PT have vastly different quality levels.  In order to claw as much performance out of the "regular" RT method they use AI reconstruction to enhance the data the probes collected but to enable it in something you need to be able to update the nVidia sdk used in the game, shader code, and that's not even really the tip of the iceberg... It doesn't seem possible but if it is the amount of time needed would be immense.

That why RTX Remix uses the path tracing method for adding the new lighting to older games and I have to imagine if they could have just done something and used ray reconstruction then nVidia would have as it's way less computationally expensive.  Adding path tracing to Max Payne for example can take a 4090 all the way down to 26 fps in scenes that are otherwise not demanding even for 15 year old GPU's.


We can always dream though... Plus who knows what tech is in the near future?

**Comment by sugarylettuce49** [comment]
Doesn't regular dlss already require motion vectors?

**Comment by d0x360** [comment]
Yeah, I whoopsed but this is kinda outdated information anyway since the way DLSS is integrated has gone from nVidia adding it to a game themselves to devs just using a plugin.. which opens avenues to mod DLSS into places it's not 'supposed' to be & that was kinda kicking off shortly after this
